pkg/detector/detector.go

```diff
@@ -700,15 +701,22 @@ func (d *ResourceDetector) GetUnstructuredObject(objectKey keys.ClusterWideKey)
 // ClaimPolicyForObject set policy identifier which the object associated with.
 func (d *ResourceDetector) ClaimPolicyForObject(object *unstructured.Unstructured, policy *policyv1alpha1.PropagationPolicy) (string, error) {
 	policyID := policy.Labels[policyv1alpha1.PropagationPolicyPermanentIDLabel]
-
+	generation := strconv.FormatInt(policy.Generation, 10)
 	objLabels := object.GetLabels()
 	if objLabels == nil {
 		objLabels = make(map[string]string)
 	} else if len(objLabels) > 0 {
 		// object has been claimed, don't need to claim again
 		if !excludeClusterPolicy(objLabels) &&
 			objLabels[policyv1alpha1.PropagationPolicyPermanentIDLabel] == policyID {
-			return policyID, nil
+			if !util.IsLazyActivationEnabled(policy.Spec.ActivationPreference) {
+				// If the object has been claimed by the same policy and the activation preference is not lazy, no need to claim again.
+				return policyID, nil
+			}
+			if object.GetAnnotations()[policyv1alpha1.PropagationPolicyGenerationAnnotation] == generation {
+				// If the object has been claimed by the same policy and the generation is the same and the activation preference is lazy, no need to claim again.
+				return policyID, nil
+			}
```

Comment:
I got your mean: if the policy is lazy and its generation is newer, we shouldn't just return, we should fallthrough to refresh the generation in objectAnnotations.

But, I still think here has some error, supposing:

1. `policyA` (lazy, gen=1), the bound resource `deployA` (claimGen=1)
2. `policyA` updated (gen=2)
3. `ClaimPolicyForObject` function called
4. meets `lazy && claimGen != gen`, so `deployA`'s claimGen=2
5. however, `deployA` didn;t modified by user, the claimGen should still keep as 1.

has you tested it?

Reply:
i have test that the claimed generation will always be latest generation of policy, which is not expected

Reply:
@chaosi-zju please take a look again.